src/main/java/io/jenkins/plugins/pipelinegraphview/PipelineGraphDisplayURLProvider.java

```diff
@@ -24,6 +25,30 @@ public String getName() {
     @Override
     @NonNull
     public String getRunURL(Run<?, ?> run) {
-        return getRoot() + Util.encode(run.getUrl()) + "pipeline-graph";
+        if (run instanceof WorkflowRun) {
+            return DisplayURLProvider.getDefault().getRunURL(run) + "pipeline-graph";
+        }
+        return DisplayURLProvider.getDefault().getRunURL(run);
+    }
+
+    @Override
+    public String getArtifactsURL(Run<?, ?> run) {
+        return DisplayURLProvider.getDefault().getArtifactsURL(run);
+    }
+
+    @Override
+    public String getChangesURL(Run<?, ?> run) {
+        return DisplayURLProvider.getDefault().getChangesURL(run);
+    }
+
+    @Override
+    public String getTestsURL(Run<?, ?> run) {
+        return DisplayURLProvider.getDefault().getTestsURL(run);
+    }
```

Comment:
Without these, artifact/test specific links end up taking us to an invalid URL related to `/pipeline-graph`, which 404s, and then takes us back to the main build page. This change allows us to go to the proper pages automatically.